iocore/net/SSLNetVConnection.cc

```diff
@@ -306,15 +306,14 @@ ssl_read_from_net(SSLNetVConnection *sslvc, EThread *lthread, int64_t &ret)
       Debug("ssl.error", "SSL_ERROR_WOULD_BLOCK(read/x509 lookup)");
       break;
     case SSL_ERROR_SYSCALL:
-      SSL_INCREMENT_DYN_STAT(ssl_error_syscall);
       if (nread != 0) {
         // not EOF
+        SSL_INCREMENT_DYN_STAT(ssl_error_syscall);
         event = SSL_READ_ERROR;
         ret   = errno;
         Debug("ssl.error", "SSL_ERROR_SYSCALL, underlying IO error: %s", strerror(errno));
       } else {
         // then EOF observed, treat it as EOS
```

Comment:
Do we want to track EOS here?

Reply:
Actually, it's my first idea. But I decided to do nothing to follow what we did for `SSL_ERROR_WANT_WRITE`, `SSL_ERROR_WANT_READ`, and `SSL_ERROR_ZERO_RETURN` with #5389.

Reply:
If we really need these stats, let's restore all of them with renaming.